File: ployglots/mta-ployglot-pkg-fetch-client/packages/ts/examples/sdk/02-caching-client.ts
// @ts-nocheck
/**
 * Example: cached client via `CachingClient`.
 *
 * `CachingClient` wraps an `AsyncClient` with a `CacheManager` so repeated GETs
 * for the same URL are served from memory. The `cache.stats()` accessor reports
 * cumulative `hits` / `misses` / `evictions` / `size`.
 */
import { CachingClient } from '@polyglot/fetch-http-client';

async function main(): Promise<void> {
  const client = new CachingClient({
    baseUrl: 'https://jsonplaceholder.typicode.com',
    cache: { ttl: 60_000 }, // 60s in-memory TTL
  });
  try {
    // First call: cache miss → network.
    await client.get('/posts/1');
    // Second call: cache hit → no network.
    await client.get('/posts/1');

    const stats = client.cache.stats();
    console.log('cache hits:', stats.hits);
    console.log('cache misses:', stats.misses);
    console.log('cache size:', stats.size);
  } finally {
    await client.close();
  }
}

main().catch((err: unknown) => {
  console.error(err);
  process.exitCode = 1;
});

File: ployglots/mta-ployglot-pkg-fetch-client/packages/py/examples/sdk/02_caching_client.py
"""SDK example: CachingClient.

The second GET for the same URL is served from memory. `cache.stats()`
reports hits / misses / sets.

Run:
    python -m examples.sdk.02_caching_client
"""

from __future__ import annotations

import asyncio
import sys

from fetch_http_client import AsyncClient, CacheConfig, CachingClient


async def main() -> None:
    async with AsyncClient(base_url="https://httpbin.org") as inner:
        async with CachingClient(inner, config=CacheConfig(ttl_seconds=60)) as client:
            await client.get("/uuid")  # miss → set
            await client.get("/uuid")  # hit
            stats = client.cache.stats().as_dict()
            sys.stdout.write(f"cache stats: {stats}\n")


if __name__ == "__main__":
    asyncio.run(main())
